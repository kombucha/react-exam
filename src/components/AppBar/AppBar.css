.AppBar {
  display: flex;
  flex-direction: row;
  align-items: center;
  height: 64px;
  padding: 8px 16px;

  background-color: #222;
  border-bottom: 1px solid #333;
  color: white;
}

.AppBar__link {
  text-decoration: none;
  color: inherit;
}

.AppBar__title {
  font-family: StarWarsHollow;
  font-size: 1.5rem;
}
