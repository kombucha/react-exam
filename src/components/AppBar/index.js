import React from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";

import "./AppBar.css";

const AppBar = ({ title }) => (
  <div className="AppBar">
    <NavLink className="AppBar__link" to="/">
      <h1 className="AppBar__title">{title} </h1>
    </NavLink>
  </div>
);

AppBar.propTypes = {
  title: PropTypes.string.isRequired,
};

export default AppBar;
