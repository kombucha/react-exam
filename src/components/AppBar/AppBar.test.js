import React from "react";
import AppBar from "./index";
import { shallow } from "enzyme";

it("should render properly", () => {
  const wrapper = shallow(<AppBar title="Oh hai" />);
  expect(wrapper).toMatchSnapshot();
});
