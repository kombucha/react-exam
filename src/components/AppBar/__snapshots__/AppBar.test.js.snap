// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppBar
    title="Oh hai"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <NavLink
        activeClassName="active"
        ariaCurrent="true"
        className="AppBar__link"
        to="/"
>
        <h1
                className="AppBar__title"
        >
                Oh hai
                 
        </h1>
</NavLink>,
      "className": "AppBar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "active",
        "ariaCurrent": "true",
        "children": <h1
          className="AppBar__title"
>
          Oh hai
           
</h1>,
        "className": "AppBar__link",
        "to": "/",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Oh hai",
            " ",
          ],
          "className": "AppBar__title",
        },
        "ref": null,
        "rendered": Array [
          "Oh hai",
          " ",
        ],
        "type": "h1",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="AppBar__link"
          to="/"
>
          <h1
                    className="AppBar__title"
          >
                    Oh hai
                     
          </h1>
</NavLink>,
        "className": "AppBar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <h1
            className="AppBar__title"
>
            Oh hai
             
</h1>,
          "className": "AppBar__link",
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Oh hai",
              " ",
            ],
            "className": "AppBar__title",
          },
          "ref": null,
          "rendered": Array [
            "Oh hai",
            " ",
          ],
          "type": "h1",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
